job-server/src/spark.jobserver/JobManagerActor.scala

```diff
@@ -215,7 +224,7 @@ class JobManagerActor(dao: JobDAO,
           }
           case SparkJobValid => {
             statusActor ! JobStarted(jobId: String, contextName, jobInfo.startTime)
-            job.runJob(sparkContext, jobConfig)
+            job.runJob(jobContext.asInstanceOf[job.C], jobConfig)
```

Comment:
I think it would be more readable if you extract the asInstanceOf to a val (val jobInstance = jobContext.asInstanceOf[job.C])